overlord/ifacestate/handlers.go

```diff
@@ -197,7 +198,19 @@ func (m *InterfaceManager) setupProfilesForSnap(task *state.Task, _ *tomb.Tomb,
 		affectedSnaps = append(affectedSnaps, name)
 	}
 	sort.Strings(affectedSnaps)
-	return m.setupAffectedSnaps(task, snapName, affectedSnaps)
+	err = m.setupAffectedSnaps(task, snapName, affectedSnaps)
+
+	// Store the snap revision in the state, so that we know which revision has profiles.
+	st := task.State()
+	var snapifst SnapInterfaceState
+	if err := Get(st, snapName, &snapifst); err != nil && err != state.ErrNoState {
+		logger.Noticef("cannot get interface state of snap %q: %s", err)
```

Comment:
Why do we Get first if our intent is to simple (over)write whatever revision is stored there?

Reply:
Because this is correct iff we add new fields there, this is a read-modify-write cycle designed to ensure that we don't clobber other data.